### DISCUSSION TITLE
what is Command in odoo 17?

### DISCUSSION THREAD

**Discussion by it12uw** [OCA/odoo-community.org#157] [Q&A] [answered]
hello everyone, i want to ask a question about odoo 17.
what is "Command" in odoo 17?

from odoo import api, fields, models, _, Command

**Comment by luisg123v** [comment, marked as answer]
Starting from Odoo 15.0, the Command syntax is the preferred way to set values to `One2many` and `Many2many` fields when creating or writing to records, instead of using numeric commands.

For instance, instead of:

```
self.write({"partner_ids": [(6, 0, partners.ids)]})
```

We would do:

```
self.write({"partner_ids": [Command.set(partners.ids)]})
```

Note that this syntax also works in XML, e.g.:

```
<field name="partner_ids" eval="[Command.set(ref('base.main_partner'))]" />
```

Here is how to translate from the old syntax to the new one:

| Numeric syntax | Command syntax |
|--------|--------|
| `(0, 0, {...})` | `Command.create({...})` |
| `(1, id, {...})` | `Command.update(id, {...})` |
| `(2, id)` | `Command.delete(id)` |
| `(3, id)` | `Command.unlink(id)` |
| `(4, id)` | `Command.link(id)` | 
| `(5, 0, 0)` | `Command.clear()` | 
| `(6, 0, ids)` | `Command.set(ids)` |

For more information, you may check [the Command class documentation](https://www.odoo.com/documentation/17.0/developer/reference/backend/orm.html#odoo.fields.Command)

I hope this helps.

**Reply by it12uw** [reply]
thankyou so much for the information
i get it now
